mava/utils/gae.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Union
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from mava.systems.ppo.types import PPOTransition, RNNPPOTransition
+
+
+def calculate_gae(
+    traj_batch: Union[PPOTransition, RNNPPOTransition],
+    last_val: chex.Array,
+    last_done: chex.Array,
+    gamma: float,
+    gae_lambda: float,
+) -> Tuple[chex.Array, chex.Array]:
```

Comment:
Now that it's in utils it needs some good comments

```suggestion
) -> Tuple[chex.Array, chex.Array]:
    """Computes truncated generalized advantage estimates.
    
    The advantages are computed in a backwards fashion according to the equation:
    Âₜ = δₜ + (γλ) * δₜ₊₁ + ... + ... + (γλ)ᵏ⁻ᵗ⁺¹ * δₖ₋₁
    where δₜ = rₜ₊₁ + γₜ₊₁ * v(sₜ₊₁) - v(sₜ).

    See Proximal Policy Optimization Algorithms, Schulman et al.:
    https://arxiv.org/abs/1707.06347
    
    Args:
        traj_batch (B, T, A, ...): a batch of trajectories.
        last_val  (B, A): value of the final timestep.
        last_done (B, A): whether the last timestep was a terminated or truncated.
        gamma (): discount factor.
        gae_lambda (): GAE mixing parameter.
    
    Returns Tuple[(B, T, A), (B, T, A)]: advantages and target values.
    """
```